test/extensions/filters/http/common/jwks_fetcher_test.cc

```diff
@@ -43,106 +40,119 @@ const char publicKey[] = R"(
 }
 )";
 
-const std::string JwksUri = R"(
-uri: https://pubkey_server/pubkey_path
-cluster: pubkey_cluster
-timeout:
-  seconds: 5
+const std::string config = R"(
+http_uri:
+  uri: https://pubkey_server/pubkey_path
+  cluster: pubkey_cluster
+  timeout:
+    seconds: 5
 )";
 
 class JwksFetcherTest : public testing::Test {
 public:
-  void SetUp() override {
-    TestUtility::loadFromYaml(JwksUri, uri_);
+  void setupFetcher(const std::string& config_str) {
+    TestUtility::loadFromYaml(config_str, remote_jwks_);
     mock_factory_ctx_.cluster_manager_.initializeThreadLocalClusters({"pubkey_cluster"});
   }
 
-  HttpUri uri_;
+  RemoteJwks remote_jwks_;
   testing::NiceMock<Server::Configuration::MockFactoryContext> mock_factory_ctx_;
   NiceMock<Tracing::MockSpan> parent_span_;
 };
 
 // Test findByIssuer
 TEST_F(JwksFetcherTest, TestGetSuccess) {
   // Setup
+  setupFetcher(config);
   MockUpstream mock_pubkey(mock_factory_ctx_.cluster_manager_, "200", publicKey);
   MockJwksReceiver receiver;
-  std::unique_ptr<JwksFetcher> fetcher(JwksFetcher::create(mock_factory_ctx_.cluster_manager_));
+  std::unique_ptr<JwksFetcher> fetcher(JwksFetcher::create(
+      mock_factory_ctx_.cluster_manager_, remote_jwks_, mock_factory_ctx_.dispatcher_));
```

Comment:
It seems every test function is calling line 66, 69,70.   Maybe they can be moved into the SetUp function?

Reply:
done